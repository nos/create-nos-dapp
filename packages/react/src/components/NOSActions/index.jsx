import React from "react";
import injectSheet from "react-jss";
import PropTypes from "prop-types";
import { injectNOS, nosProps } from "@nosplatform/api-functions/lib/react";

const styles = {
  button: {
    margin: "16px",
    fontSize: "14px"
  }
};

class NOSActions extends React.Component {
  handleAlert = async func => alert(await func);

  // handleGetAddress = async () => alert(await this.props.nos.getAddress());

  handleClaimGas = () =>
    this.props.nos
      .claimGas()
      .then(alert)
      .catch(alert);

  render() {
    const { classes, nos } = this.props;

    const neo = "c56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b";
    const gas = "602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7";
    const rpx = "ecc6b20d3ccac1ee9ef109af5a7cdb85706b1df9";

    // Add your smart contract's scriptHash here
    const scriptHash = "";

    // The operation of your smart contract you want to (test)invoke
    const operation = "";

    // The necessary arguments for you (test)invoke
    const args = [];

    // The storagekey you want to query
    const key = "";

    // The asset, amount and receiver of your send function
    const receiver = "";
    const amount = "";
    const asset = gas;

    const invoke = { scriptHash, operation, args }; // and testInvoke
    const getStorage = { scriptHash, key };
    const send = { amount, asset, receiver };

    return (
      <React.Fragment>
        <button className={classes.button} onClick={() => this.handleAlert(nos.getAddress())}>
          Get Address
        </button>
        <button
          className={classes.button}
          onClick={() => this.handleAlert(nos.getBalance({ asset: neo }))}
        >
          Get NEO Balance
        </button>
        <button
          className={classes.button}
          onClick={() => this.handleAlert(nos.getBalance({ asset: gas }))}
        >
          Get GAS Balance
        </button>
        <button
          className={classes.button}
          onClick={() => this.handleAlert(nos.getBalance({ asset: rpx }))}
        >
          Get RPX Balance
        </button>

        <button className={classes.button} onClick={this.handleClaimGas}>
          Claim Gas
        </button>
        <button className={classes.button} onClick={() => this.handleAlert(nos.send(send))}>
          Send GAS to...
        </button>

        <button className={classes.button} onClick={() => this.handleAlert(nos.testInvoke(invoke))}>
          TestInvoke
        </button>
        {/*
          <button
            className={classes.button}
            onClick={() => this.handleAlert(nos.invoke(invoke))}
          >
            Invoke
          </button>
        */}
        <button
          className={classes.button}
          onClick={() => this.handleAlert(nos.getStorage(getStorage))}
        >
          GetStorage
        </button>
      </React.Fragment>
    );
  }
}

NOSActions.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  nos: nosProps.isRequired
};

export default injectNOS(injectSheet(styles)(NOSActions));
