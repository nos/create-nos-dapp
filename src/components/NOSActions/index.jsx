import React from "react";
import injectSheet from "react-jss";
import PropTypes from "prop-types";
import { injectNOS, NOSTypes } from "../../nos";

const styles = {
  button: {
    margin: "16px",
    fontSize: "14px"
  }
};

const fallback = () => "You're not working in nOS!";

class NOSActions extends React.Component {
  handleGetAddress = async () =>
    alert(await this.props.nos.getAddress({ fallback }));

  handleClaimGas = () =>
    this.props.nos
      .claimGas({ fallback })
      .then(alert)
      .catch(alert);

  handleGetBalance = async scriptHash =>
    alert(await this.props.nos.getBalance(scriptHash, { fallback }));

  handleTestInvoke = async (scriptHash, operation, args) =>
    alert(
      await this.props.nos.testInvoke(scriptHash, operation, args, { fallback })
    );

  handleInvoke = async (scriptHash, operation, args) =>
    alert(
      await this.props.nos.testInvoke(scriptHash, operation, args, { fallback })
    );

  handleGetStorage = async (scriptHash, key) =>
    alert(await this.props.nos.getStorage(scriptHash, key, { fallback }));

  render() {
    const { classes } = this.props;

    // Get Balance
    const neo =
      "c56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b";
    // const gas = "602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7";
    // const rpx = "ecc6b20d3ccac1ee9ef109af5a7cdb85706b1df9";

    // (test) Invoke
    const neoAuth = "2f228c37687d474d0a65d7d82d4ebf8a24a3fcbc";
    const operation = "9937f74e-1edc-40ae-96ad-1120166eab1b";
    const args = "ef68bcda-2892-491a-a7e6-9c4cb1a11732";

    // Get Storage
    const neoBlog = "85e9cc1f18fcebf9eb8211a128807e38d094542a";
    const key = "post.latest";

    return (
      <React.Fragment>
        <button className={classes.button} onClick={this.handleGetAddress}>
          Get Address
        </button>
        <button
          className={classes.button}
          onClick={() => this.handleGetBalance(neo)}
        >
          Get NEO Balance
        </button>
        {/*
          <button
            className={classes.button}
            onClick={() => this.handleGetBalance(gas)}
          >
            Get GAS Balance
          </button>
          <button
            className={classes.button}
            onClick={() => this.handleGetBalance(rpx)}
          >
            Get RPX Balance
          </button>
        */}
        <button className={classes.button} onClick={this.handleClaimGas}>
          Claim Gas
        </button>
        <button
          className={classes.button}
          onClick={() => this.handleTestInvoke(neoAuth, operation, args)}
        >
          TestInvoke (NeoAuth)
        </button>
        {/*
          <button
            className={classes.button}
            onClick={() => this.handleInvoke(neoAuth, operation, args)}
          >
            Invoke (NeoAuth)
          </button>
        */}
        <button
          className={classes.button}
          onClick={() => this.handleGetStorage(neoBlog, key)}
        >
          GetStorage (NeoBlog)
        </button>
      </React.Fragment>
    );
  }
}

NOSActions.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  nos: NOSTypes.isRequired
};

export default injectNOS(injectSheet(styles)(NOSActions));
