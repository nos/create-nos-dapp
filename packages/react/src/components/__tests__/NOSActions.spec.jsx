import React from "react";
import { render } from "enzyme";
import NOSActions from "./../NOSActions";

describe("Components", () => {
  describe("NOSActions", () => {
    test("should render", () => {
      const wrapper = render(<NOSActions />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
