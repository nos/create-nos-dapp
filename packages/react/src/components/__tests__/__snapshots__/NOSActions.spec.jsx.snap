// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components NOSActions should render 1`] = `
Array [
  <button
    class="NOSActions-button-0-1-1"
  >
    Get Address
  </button>,
  <button
    class="NOSActions-button-0-1-1"
  >
    Get NEO Balance
  </button>,
  <button
    class="NOSActions-button-0-1-1"
  >
    Claim Gas
  </button>,
  <button
    class="NOSActions-button-0-1-1"
  >
    TestInvoke (NeoAuth)
  </button>,
  <button
    class="NOSActions-button-0-1-1"
  >
    GetStorage (NeoBlog)
  </button>,
]
`;
